cpp/modules/deck.gl/core/src/viewports/web-mercator-viewport.cpp

```diff
@@ -98,7 +98,59 @@ auto WebMercatorViewport::getMapCenterByLngLatPosition(mathgl::Vector2<double> l
   return this->unprojectFlat(newCenter);
 }
 
-auto fitBounds(mathgl::Vector2<double> topLeft, mathgl::Vector2<double> bottomRight, int padding = 0,
-               mathgl::Vector2<int> offset = mathgl::Vector2<int>()) {
-  throw new std::logic_error("fitBounds not supported");
+auto WebMercatorViewport::fitBounds(mathgl::Vector2<double> topLeft, mathgl::Vector2<double> bottomRight, int padding,
+                                    mathgl::Vector2<int> offset) -> WebMercatorViewport {
+  // TODO(randy@unfolded.ai)
+  // https://github.com/uber-web/math.gl/blob/master/modules/web-mercator/src/fit-bounds.js
+
+  WebMercatorViewport::Options options;
+  options.width = this->width;
+  options.height = this->height;
+  options.longitude = 0;
+  options.latitude = 0;
+  options.zoom = 0;
+
+  // Assume constant padding for now
+  // Call WebMercatorViewport constructor, passing in opts
+  WebMercatorViewport viewport = WebMercatorViewport(options);
+
+  auto nw = viewport.projectFlat(topLeft);
+  auto se = viewport.projectFlat(bottomRight);
+
+  // get abs of x/y difference
+  // max(abs, minExtent)?
+  const double size[] = {abs(se.x - nw.x), abs(se.y - nw.y)};
+
+  const double targetSize[] = {this->width - 2 * padding - abs(offset.x) * 2,
+                               this->height - 2 * padding - abs(offset.y) * 2};
+
+  // mimicing assert(); in JS
+  if (targetSize[0] <= 0 || targetSize[1] <= 0) {
+    throw new std::logic_error("Invalid targetSize");
+  }
+
+  const double scaleX = targetSize[0] / size[0];
+  const double scaleY = targetSize[1] / size[1];
+
+  // From JS:
+  //    Find how much we need to shift the center
```

Comment:
If padding is set, maybe throw the not implemented error for now?

Reply:
For this issue, should padding be implemented as in the JS implementation? 

In other words, should there be separate padding values for each of the cardinal directions or a single padding value for all directions as in the .cpp declaration (`int padding`)?

Reply:
On a similar note, should `maxZoom` and `minExtent` also be included as input parameters?